packages/avatar/src/vaadin-avatar-mixin.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { FocusMixin } from '@vaadin/a11y-base/src/focus-mixin.js';
+
+/**
+ * A mixin providing common avatar functionality.
+ *
+ * @polymerMixin
+ * @mixes FocusMixin
+ */
+export const AvatarMixin = (superClass) =>
+  class AvatarMixinClass extends FocusMixin(superClass) {
+    static get properties() {
+      return {
+        /**
+         * The path to the image
+         */
+        img: {
+          type: String,
+          reflectToAttribute: true,
+          observer: '__imgChanged',
+        },
+
+        /**
+         * A shortened form of name that is displayed
+         * in the avatar when `img` is not provided.
+         */
+        abbr: {
+          type: String,
+          reflectToAttribute: true,
+        },
+
+        /**
+         * Full name of the user
+         * used for the tooltip of the avatar.
+         */
+        name: {
+          type: String,
+          reflectToAttribute: true,
+        },
+
+        /**
+         * Color index used for avatar background.
+         * @attr {number} color-index
+         */
+        colorIndex: {
+          type: Number,
+          observer: '__colorIndexChanged',
+        },
+
+        /**
+         * The object used to localize this component.
+         * To change the default localization, replace the entire
+         * _i18n_ object or just the property you want to modify.
+         *
+         * The object has the following JSON structure and default values:
+         *
+         * ```
+         * {
+         *   // Translation of the anonymous user avatar tooltip.
+         *   anonymous: 'anonymous'
+         * }
+         * ```
+         *
+         * @type {!AvatarI18n}
+         * @default {English/US}
+         */
+        i18n: {
+          type: Object,
+          value: () => {
+            return {
+              anonymous: 'anonymous',
+            };
+          },
+        },
+
+        /**
+         * When true, the avatar has tooltip shown on hover and focus.
+         * The tooltip text is based on the `name` and `abbr` properties.
+         * When neither is provided, `i18n.anonymous` is used instead.
+         * @attr {boolean} with-tooltip
+         */
+        withTooltip: {
+          type: Boolean,
+          value: false,
+          observer: '__withTooltipChanged',
+        },
+
+        /** @protected */
+        __imgVisible: Boolean,
```

Comment:
Changed the visibility of properties / methods used in the template to `protected`.